app/lib/package/backend.dart

```diff
@@ -1156,14 +1169,38 @@ class PackageBackend {
           'publishing is only allowed from "push" events, this token originates from a "${agent.payload.eventName}" event');
     }
 
+    if (agent.payload.refType != 'tag') {
+      throw AuthorizationException.githubActionIssue(
+          'publishing is only allowed from "tag" refType, this token has "${agent.payload.refType}" refType');
+    }
+    final expectedRefStart = 'refs/tags/';
+    if (!agent.payload.ref.startsWith(expectedRefStart)) {
+      throw AuthorizationException.githubActionIssue(
+          'publishing is only allowed from "refs/tags/*" ref, this token has "${agent.payload.ref}" ref');
+    }
+    final tagPattern = githubPublishing.tagPattern ?? '{{version}}';
+    if (!tagPattern.contains('{{version}}')) {
```

Comment:
This check seems redundant given the validation above. Maybe it is wise to keep it...

Reply:
Yeah, in theory we shouldn't have this exception triggered, however, I'd rather have it there, in case we screw up e.g. and admin API that allows updating it

Reply:
This is not an authorization error, it's a 500 error. One could argue misconfiguration, but if we're changing the format then everything configured before should be updated or disabled by us in a backfill job.

So this seems like an internal error to me.

Reply:
Updated to `ArgumentError`.